### PR TITLE
All claims 31 mg

### PR DESCRIPTION
**Pull request by markgreenburg**
## Description
Adds a supporting evidence chapter with intro page, and second page with ability to select evidence types that will be submitted with the form.

_Note:_ Separate PRs will address the other pages noted in the attached ticket.

## Testing done
- Tested locally
- Added unit tests

## Screenshots

Chapter orientation page:
screen shot 2018-09-20 at 1 45 27 pm

Base screen for second page:
screen shot 2018-09-20 at 12 23 37 pm

-----

If veteran selects 'no evidence':
screen shot 2018-09-20 at 12 23 50 pm

-----

If veteran selects 'yes' for evidence:
screen shot 2018-09-20 at 12 23 59 pm

-----

If the veteran clicks on the evidence types helper link:
screen shot 2018-09-20 at 1 41 36 pm

## Acceptance criteria
- [x] Veterans are able to select types of evidence they want to submit with their claim
- [x] We require them to select an evidence type if they say they have evidence to submit

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by peggygannon — CHANGES_REQUESTED** [review]
@markgreenburg Please see below for my edits for this section: 
1. I've edited the **Please note:** box a bit to cut down on the repetitive wording. 
Also, do you think this content should be in a blue box to stand out a bit? 
**Please note:** You don’t have to submit evidence for your claim, but we recommend that you do provide some supporting information related to your claimed disability. If you don't submit any evidence, we may schedule a claim exam for you to help us decide your claim.<br>
You have up to 1 year from the date we receive your claim to turn in any evidence. If you don't have supporting evidence right now, you can save your application and return to it later when your evidence is ready to upload. 


2. Please update the following content to match the current outline (Benefits should be plural)
**Private medical records**
If you were treated by a private doctor, including a Veteran’s Choice doctor, you’ll have private medical records. We’ll need to see those records to make a decision on your claim. A Disability Benefits Questionnaire is an example of a private medical record.

3. Please update the following content to match the current outline: 
**Lay statements or other evidence**
A lay statement is a written statement from family, friends, or coworkers to help support your claim. Lay statements are also called “buddy statements.” In most cases, you’ll only need your medical records to support your disability claim. But some claims—such as those for Posttraumatic Stress Disorder or military sexual trauma—could benefit from a lay or buddy statement.

File: src/applications/disability-benefits/all-claims/pages/evidenceTypes.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      'view:hasPrivateMedicalRecords': { 'ui:title': 'Private medical records' },
+      'view:hasOtherEvidence': { 'ui:title': 'Supporting (lay) statements or other evidence' }
+    },
+    'view:evidenceTypeHelp': {
+      'ui:title': ' ',
+      'ui:description': evidenceTypeHelp
+    }
+  },
+  'view:noEvidenceFollowUp': {
+    'ui:title': ' ',
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/pages/evidenceTypes.js]: I'm sure this is needed, but I forget why. Do we normally add a title of `' '` when we don't want to give an element a title, but want it to create a label so we can show validation errors on it? Is there another reason for this?

Ideally, we can express why we do this so we can file a ticket in USFS...so we don't _have_ to do this anymore.

File: src/applications/disability-benefits/all-claims/tests/pages/evidenceTypes.unit.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        schema={schema}
+        uiSchema={uiSchema}
+        data={{}}
+        formData={{}}/>
+    );
+
+    expect(form.find('input').length).to.equal(2);
+  });
+
+  it('should require selecting evidence or no evidence', () => {
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/tests/pages/evidenceTypes.unit.spec.jsx]: Ohh, I get it now. Took me a while. What do you think of "should require selecting an option?"

File: src/applications/disability-benefits/all-claims/validations.js [hunk 3]
```diff
@@ -44,3 +44,10 @@ export function validateMilitaryState(errors, state, formData, schema, messages,
     errors.addError('State must be AA, AE, or AP when using a military city');
   }
 }
+
+export const validateBooleanIfEvidence = (errors, fieldData, formData, schema, messages, options, index) => {
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/validations.js]: :thinking: So we run the normal validations if we have evidence.

As we're using it in conjunction with USFS' `validateBooleanGroup`, this means we require an evidence type to be selected _if_ we've previously selected `yes` to the `do you have any evidence` question, right?

Took me way too long to figure that one out. :grimacing:

What do you think of `validateIfHasEvidence` or something? (Still not a great name, I know.) This function isn't strictly speaking a boolean validator--that's the `wrappedValidator` we're passing.

**Review by cvalarida — APPROVED** [review]
Looks good. I had a few questions and comments, but nothing blocking.

File: src/applications/disability-benefits/all-claims/tests/pages/evidenceTypes.unit.spec.jsx (hunk 2 above)
Comment (markgreenburg) [reply src/applications/disability-benefits/all-claims/tests/pages/evidenceTypes.unit.spec.jsx]: Sometimes words are hard.

**Comment by markgreenburg** [comment]
@peggygannon please see below screenshots with the new content. Thanks for the review!

screen shot 2018-09-24 at 4 39 35 pm

-----

screen shot 2018-09-24 at 4 47 52 pm

**Comment by peggygannon** [comment]
@markgreenburg Hi, Mark... just two small tweaks

1. Please add But in this paragraph: (I've bolded it so you can see where to put it)

Lay statements or other evidence
A lay statement is a written statement from family, friends, or coworkers to help support your claim. Lay statements are also called “buddy statements.” In most cases, you’ll only need your medical records to support your disability claim. **But** some claims—such as those for Posttraumatic Stress Disorder or military sexual trauma—could benefit from a lay or buddy statement.


2. Under **Please note:** Please remove comma from this sentence: 
If you don't submit any evidence we may schedule a claim exam for you to help us decide your claim.

**Comment by markgreenburg** [comment]
@peggygannon sorry, looks like I missed those. Here're a couple new screenshots:

screen shot 2018-09-25 at 4 42 11 pm

screen shot 2018-09-25 at 4 41 46 pm

**Review by peggygannon — APPROVED** [review]

**Pull request merged**